source/composer/statistics.cpp

```diff
@@ -0,0 +1,28 @@
+#include "statistics.h"
+#include "optimizer.hpp"
+
+namespace composer {
+    std::vector<melody> find_best_individuals(optimizer<nsga_ii>& ga) {
```

Comment:
"best" parece meio estranho aqui. O otimizador teoricamente já encontrou os melhores indivíduos. O que vc quer aqui são os indivíduos mais próximos de alguns pontos. E já existe o find_nearest para isso.

Como a função tem só um parâmetro e basicamente chama a função `find_nearest`, não sei de realmente precisamos de uma função para isso. No momento que precisar disso na interface, vc chama o find_nearest para os valores que precisar.

Se sempre for precisar dos mesmos valores, daí o mais provável é que não precise é do nsga_ii, pois já sabe que a função de otimização deveria estar indo em uma direção. Ou pode rodar os algoritmos para com vetores específicos de direções (MOEA/D se não me engano).